policy/lib/image.rego

```diff
@@ -28,6 +28,33 @@ parse(ref) = d {
 	}
 }
 
+# equal_ref returns if two image references point to the same image. The
+# algorithm first checks if the constituent parts repository, tag and digest are
+# all equal, if they're not the tag is ignored and repository and digest must be
+# equal
+equal_ref(ref1, ref2) {
+	img1 := parse(ref1)
+	img2 := parse(ref2)
+
+	img1 == img2
+}
+
+equal_ref(ref1, ref2) {
+	img1 := parse(ref1)
+	img2 := parse(ref2)
+
+	img1.digest != ""
+	object.remove(img1, ["tag"]) == object.remove(img2, ["tag"])
+}
+
+equal_ref(ref1, ref2) {
```

Comment:
I don't see why this block is needed. I commented it out and the tests still pass but code coverage is still 100% :thinking: 

I think it might be impossible for line 55 to ever evaluate to true. The evaluations on line 39 and line 47 capture all the possible scenarios.